### PR TITLE
Shared auth session between app.near.ai and chat.near.ai

### PR DESCRIPTION
**Pull request by calebjacob**
Closes: https://github.com/nearai/nearai/issues/705

- Ended up needing to clean up our tRPC + React Query configuration to properly support reading and setting cookies in our tRPC middleware layer.
- Migration logic to keep users signed in. This moves their auth token from local storage to a `HttpOnly`, `Secure` cookie scoped to `Domain=near.ai` in production. This will make the cookie accessible for `near.ai` and `*.near.ai` domains.
- ENV variable updated for `app.near.ai` and `chat.near.ai` => `AUTH_COOKIE_DOMAIN=near.ai`
- All other environments (staging, preview builds) won't have a value set for `AUTH_COOKIE_DOMAIN` - meaning the cookie will be scoped only to the exact domain its running on.
- Added a little README for future reference: `hub/demo/src/trpc/README.md`

I've done my best to thoroughly test all the edge cases I can think of and it seems to work smoothly.

### CODE REVIEW

File: hub/demo/src/components/Footer.tsx [hunk 1]
```diff
@@ -46,7 +46,7 @@ export const Footer = ({ conditional }: Props) => {
             size="text-xs"
             color="sand-11"
           >
-            near.ai
+            NEAR.AI
```
Comment (arktoswb) [inline hub/demo/src/components/Footer.tsx:49]: Should it be "NEAR AI"?

**Review by arktoswb — APPROVED** [review]

File: hub/demo/src/components/Footer.tsx (hunk 1 above)
Comment (calebjacob) [reply hub/demo/src/components/Footer.tsx]: Good catch - fixed!
